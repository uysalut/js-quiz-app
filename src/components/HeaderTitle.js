import React from 'react';
import { View, Text, StyleSheet, Dimensions } from 'react-native';

const { width, height } = Dimensions.get('window');

const HeaderTitle = ({ title }) => {
  return (
    <View style={styles.container}>
      <Text numberOfLines={1} style={styles.title}>
        {title}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    maxWidth: width * 0.6
  },
  title: {
    fontSize: height < 650 ? 20 : 22,
    paddingTop: height < 650 ? 0 : 8,
    fontWeight: '900',
    color: '#fff',
    fontFamily: 'Avenir'
  }
});

export default HeaderTitle;
