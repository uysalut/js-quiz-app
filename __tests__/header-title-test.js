// __tests__/header-title-test.js
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HeaderTitle from '../src/components/HeaderTitle';

it('renders correctly', () => {
  const tree = renderer.create(<HeaderTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});
