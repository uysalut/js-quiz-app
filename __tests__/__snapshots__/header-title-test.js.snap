// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "maxWidth": 450,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Avenir",
        "fontSize": 22,
        "fontWeight": "900",
        "paddingTop": 8,
      }
    }
  />
</View>
`;
